xen-4.0.1/extras/mini-os/lib/math.c generates the warning
'tmp.ul[1]' may be used uninitialized in this function
under gcc 4.5 which I think is incorrect
--- xen-4.0.1/extras/mini-os/minios.mk.orig	2010-02-02 20:43:00.000000000 +0000
+++ xen-4.0.1/extras/mini-os/minios.mk	2010-07-24 22:56:27.000000000 +0100
@@ -10,6 +10,7 @@
 DEF_CFLAGS += $(call cc-option,$(CC),-fno-stack-protector,)
 DEF_CFLAGS += $(call cc-option,$(CC),-fgnu89-inline)
 DEF_CFLAGS += -Wstrict-prototypes -Wnested-externs -Wpointer-arith -Winline
+DEF_CFLAGS += -Wno-uninitialized
 DEF_CPPFLAGS += -D__XEN_INTERFACE_VERSION__=$(XEN_INTERFACE_VERSION)
 
 DEF_ASFLAGS += -D__ASSEMBLY__
